go/exorelay/features/sending.feature

```diff
@@ -0,0 +1,49 @@
+Feature: Sending outgoing messages
+
+  As an Exoservice developer
+  I want my service to be able to send messages to other Exosphere services
+  So that it can interact with the rest of the application.
+
+  Rules:
+  - call "Send" on your ExoRelay instance have it send out the given message
+  - provide the message to send as the first parameter
+  - provide payload for the message as the second parameter
+  - the payload can be either a string, an array, or a Hash
+
+
+  Background:
+    Given an ExoRelay with the role "test-service"
+    And ExoRelay connects to Exocom
+
+
+  Scenario: sending a message without payload
+    When sending the message "hello-world"
+    Then ExoRelay makes the WebSocket request:
+      """
+      {
+        "name": "hello-world",
+        "sender": "test-service",
+        "id": "{{request_uuid}}"
+      }
+      """
+
+  Scenario: sending a message with a populated Hash as payload
+    When sending the message "hello" with the payload:
+      """
+      { "name": "world" }
+      """
+    Then ExoRelay makes the WebSocket request:
+      """
+      {
+        "name": "hello",
+        "payload": {
+          "name": "world"
+        },
+        "sender": "test-service",
+        "id": "{{request_uuid}}"
+      }
+      """
+
+  Scenario: trying to send an empty message
+    When trying to send an empty message
+    Then ExoRelay errors with "ExoRelay#Send cannot send empty messages"
```

Comment:
Far fewer features here then the JS tests since a lot of them were for type checking. The JS version does allow payload to be a string or an array. That seems harder with types. Is that a requirement?